scripts/prune_lm_dir.py

```diff
@@ -328,6 +351,153 @@ def FinalizeOutput(final_work_out):
         os.remove(args.lm_dir_out + "/num_splits")
 
 
+def MatchTargetSize(num_ngrams):
+    assert(args.target_num_ngrams > 0)
+    return abs(num_ngrams - args.target_num_ngrams) / float(args.target_num_ngrams) < args.tolerance
+
+def IterateOnce(scale, step, n_iter, recovery_step, recovery_size):
+    """Prune with a specific threshold and check whether the resulting LM matches the target-num-ngrams
+
+    Args:
+        scale: the threshold equals to scale * args.threshold
+        step: number of current step
+        n_iter: number of current iteration
+        recovery_step, recovery_size: These two variables store the number of step and num-grams
+            for LM that is most recent and smallest but larger than target-num-ngrams. The following
+            step should prune from this LM.
+    Returns:
+        A tuple with the first element is a bool variable indicating whether we already got the right threshold.
+        And the remaining are updated value for (step, n_iter, recovery_step, recovery_size)
+    Example Usage:
+
+        step = 0
+        recovery_step = step
+        recovery_size = 0
+        n_iter = 0
+
+        threshold = init_threshold()
+        while True:
+            threshold = EstimateThreshold(args.target_num_ngrams)
+            scale = threshold / args.threshold
+
+            (succeed, step, n_iter, recovery_step, recovery_size) = IterateOnce(scale, step, n_iter, recovery_step, recovery_size)
```

Comment:
please change 'succeed' to 'success', and 'n_iter' to 'num_iters'; and use a backslash to shorten this very long line.